### DISCUSSION TITLE
How to set timezone of rabbitmq_tracing?

### DISCUSSION THREAD

**Discussion by charnet1019** [rabbitmq/rabbitmq-server#3923] [Questions]
MQ Version: 3.8.9

The rabbitmq_tracing install successed and grap the logs. But the log time is 8 hours away from the local time, i can not find how to set it. Wanted help, thanks.

**Comment by davidydang** [comment]
I met the same issue. I used 3.9.11. 
Look forward to the expert's help!

**Comment by lukebakken** [comment]
The tracing plugin gets the OS system time using this function - https://www.erlang.org/doc/apps/erts/time_correction.html#OS_System_Time

So my guess is that your system time zone is set to UTC.

**Reply by davidydang** [reply]
I check the time zone of my server. It is CST. But the tracing show UTC. Is there anything wrong with the tracing setting?

This is my linux server time setting: 
       Local time: Fri 2022-03-11 17:53:43 CST
 Universal time: Fri 2022-03-11 09:53:43 UTC
          RTC time: Fri 2022-03-11 09:53:43
        Time zone: Asia/Shanghai (CST, +0800)
System clock synchronized: yes
      NTP service: active
 RTC in local TZ: no

**Reply by lukebakken** [reply]
Never mind, `os:system_time` will always be UTC. There is no way at this time to change this behavior.

**Reply by davidydang** [reply]
Thank you, Luke. The thing is RabbitMQ 3.1.1 can present the local time for the tracing file.

**Reply by lukebakken** [reply]
> The thing is RabbitMQ 3.1.1 can present the local time for the tracing file

This is how the timestamp is formatted in RabbitMQ 3.1.1, and it does take time zone into account:

https://github.com/rabbitmq/rabbitmq-management/blob/rabbitmq_v3_1_1/src/rabbit_mgmt_format.erl#L132-L136

Here is where the change was made to use UTC time:

https://github.com/rabbitmq/rabbitmq-management/commit/0e8a51205ed389cf5ecf6cd389411d5d0ec2bbe2

I will bring this up with the team.

**Comment by lukebakken** [comment]
@davidydang thanks for your report -

https://github.com/rabbitmq/rabbitmq-server/issues/4276

https://github.com/rabbitmq/rabbitmq-server/pull/4277
